src/include/duckdb/storage/table/table_index_list.hpp

```diff
@@ -29,6 +30,20 @@ class TableIndexList {
 			}
 		}
 	}
+
+	//! Scan the indexes, invoking the callback method for every bound entry of a specific type
+	template <class T, class FUNC>
+	void ScanBound(FUNC &&callback) {
+		lock_guard<mutex> lock(indexes_lock);
+		for (auto &index : indexes) {
+			if (index->IsBound() && T::TYPE_NAME == index->index_type) {
```

Comment:
Should we perhaps also initialize/bind indexes here? At least for the `ScanBound` in the ART filter pushdown it makes sense to instantiate the ART indexes there imo. This could also be done using a separate method of course.